crates/editor/src/editor.rs

```diff
@@ -4690,37 +4714,55 @@ impl Editor {
             let text = buffer
                 .text_for_range(start_point..end_point)
                 .collect::<String>();
+
             let mut lines = text.split("\n").collect_vec();
 
-            let lines_len = lines.len();
+            let len_before = lines.len();
             callback(&mut lines);
-
-            // This is a current limitation with selections.
-            // If we wanted to support removing or adding lines, we'd need to fix the logic associated with selections.
-            debug_assert!(
-                lines.len() == lines_len,
-                "callback should not change the number of lines"
-            );
+            let len_after = lines.len();
 
             edits.push((start_point..end_point, lines.join("\n")));
-            let start_anchor = buffer.anchor_after(start_point);
-            let end_anchor = buffer.anchor_before(end_point);
 
-            // Make selection and push
+            // Selections must change based on added and removed line count
+            let start_row = start_point.row + added_lines as u32 - removed_lines as u32;
+            let end_row = start_row + (len_after as u32) - 1;
```

Comment:
I think we'd better do a `staturating_sub(1)` here since this code is generic and does not know which action had called it.

Now it works due to line deduplication always keeping at least one line, but if somebody adds an action that removes all lines, we might get into the situation where the selection started from 0, only removed lines and left nothing in the editor.